### ISSUE TITLE
Configuration issue when running './manage start'

### ISSUE THREAD

**Issue by shader** [bcgov/permitify#32]
I attempted to follow the basic instructions in the readme, but got the following error in the docker-compose output logs after the `./manage start` step:

```
bc_registries_1                        | Copying template directory; /app/site_templates/bc_registries ...                                                                                                     
bc_registries_1                        | cp: cannot create regular file './config.toml': Permission denied                                                                                                     
bc_registries_1                        | cp: cannot create regular file './schemas.json': Permission denied                                                                                                    
bc_registries_1                        | Starting server ...
bc_registries_1                        | ERROR 2018-01-30 20:04:53,626 config 1 140171157370624 Could not find config.toml. Exiting.                                                                           
```

**Comment by shader** [comment]
I seem to have been able to bypass this error by commenting out the following line:
https://github.com/bcgov/permitify/blob/c02d53666295e559ff1ae6eb90c342f3e250b66a/docker/permitify/Dockerfile#L19

I have no idea what other repercussions this may have

**Comment by WadeBarnes** [comment]
Commenting out that line has ramifications for running in OpenShift.  Basically your running the commands as root rather than a regular user.

What OS are you using.

@nrempel, I think we need to set the permissions on the target folder so the scripts can copy the templates.  Would you agree?

**Comment by nrempel** [comment]
@WadeBarnes I think changing the permissions on the target folder should fix this issue without breaking the Openshift deployment.

@shader for now, commenting that out shouldn't affect you if you are not using Openshift.

**Comment by shader** [comment]
Yes, I'm running it locally. I'm just trying to get the demo working, and I thought that would be the easiest way.

Commenting out the line does have the unfortunate side-effect of placing root-owned files in the src directory. But I'm not sure that using id 10001 as the default user makes any more sense for local deployment; for example, my user (which owns the rest of the files in src) has ID 1000. ID 10001 doesn't even exist on my system.

**Comment by WadeBarnes** [comment]
@shader, What OS are you running docker on?  I'm assuming some flavor of Linux.

The selection of 10001 is meant to not correspond to an existing account, since OpenShift assigns a random user to the running pod.  So the purpose is to mimic that behavior.

**Comment by nrempel** [comment]
@cywolf Wade mentioned you had similar issues to this? Are you running Linux?

**Comment by shader** [comment]
Yes, I'm running a flavor of Linux. If the intent is to use a non-existent user ID, then I suppose it should be fine. There's still a non-zero chance of collision, but I'm not sure if that would be an issue.

Changing the src directory permissions might work. Two other possible solutions: 

1. You could have the `manage` script copy the configuration files, instead of the `docker-entrypoint.sh` script. Then it would be handled by the user running the manage script, who presumably owns the src directory.

2. I was also able to load the configuration by using the docker-compose volume configuration to load the config.toml and schemas.json files directly.

I switched away from the latter to commenting out the USER line because I ran into other permissions-related bugs., so they're probably not a complete solutions.

**Comment by shader** [comment]
With just `chmod a+rw` on `permitify/src`, I now get the following error instead:

```
bc_registries_1 | INFO|indy::commands::signus | src/commands/signus.rs:159 | CreateAndStoreMyDid command received                                                     
bc_registries_1 | INFO|indy::services::pool   | src/services/pool/mod.rs:891 | Sending "pi"                                                                             
bc_registries_1 | ERROR|indy::errors::indy    | src/errors/indy.rs:63  | Casting error to ErrorCode: Invalid library state: Unexpected SQLite error: attempt to write a readonly database                                                                                  
bc_registries_1 | WARNING 2018-01-30 22:07:39,572 libindy 19 140569172309760 _indy_loop_callback: Function returned error 112
```

**Comment by nrempel** [comment]
What version of Docker are you running?

**Comment by shader** [comment]
Docker version 17.09.1-ce, build 19e2cf6
docker-compose version 1.18.0, build 8dd22a9

**Comment by shader** [comment]
Oops, that last error is probably because I ran it as root once and made a root sqlite.db file in the bc_registries_wallet volume. I dropped that volume and no longer run into that error.

**Comment by nrempel** [comment]
What are the permissions of the project directory on your host machine?

What is the output of `ls -la`?

Try commenting out lines 17 and 18 (https://github.com/bcgov/permitify/blob/master/docker/docker-compose.yml#L17) in docker-compose.yml then build and try again.
